File: python/7kyu/high_and_low.py
import codewars_test as test


def high_and_low(numbers: str) -> str:
    low = None
    high = None
    for character in numbers.split(" "):
        number = int(character)
        if low is None and high is None:
            low = number
            high = number
        elif number < low:
            low = number
        elif number > high:
            high = number
    return f"{high} {low}"


# def high_and_low(numbers: str) -> str:
#     numbers = tuple(int(n) for n in numbers.split(" "))
#     return f"{max(numbers)} {min(numbers)}"


@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(high_and_low(
            "8 3 -5 42 -1 0 0 -9 4 7 4 -4"), "42 -9")
        test.assert_equals(high_and_low("1 2 3"), "3 1")

File: typescript/7kyu/high_and_low/high_and_low.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Kata = void 0;
class Kata {
    static highAndLow(numbers) {
        let intNumbers = numbers.split(" ").map(c => parseInt(c));
        return `${Math.max(...intNumbers)} ${Math.min(...intNumbers)}`;
    }
}
exports.Kata = Kata;
const chai_1 = require("chai");
describe("Example Tests", function () {
    it("Sample Test", function () {
        chai_1.assert.strictEqual(Kata.highAndLow("8 3 -5 42 -1 0 0 -9 4 7 4 -4"), "42 -9");
    });
});
